framework/include/base/NonlinearSystem.h

```diff
@@ -457,8 +457,19 @@ class NonlinearSystem : public SystemTempl<TransientNonlinearImplicitSystem>,
    */
   virtual Real relativeSolutionDifferenceNorm();
 
+  virtual NumericVector<Number> & solution() override { return *_sys.solution; }
+
+  virtual NumericVector<Number> & solutionOld() override { return *_sys.old_local_solution; }
+
+  virtual NumericVector<Number> & solutionOlder() override { return *_sys.older_local_solution; }
+
+  virtual TransientNonlinearImplicitSystem & sys() { return _sys; }
+
+  virtual System & system() override { return _sys; }
```

Comment:
I gotta say - I'm not too happy about having `sys()` and `system()`... both with different return types.  I understand the reason for it... but I don't like the naming... I'll always forget which is which...